docs/docs/developers/contracts/references/storage/main.md

```diff
@@ -1,481 +1,115 @@
+## Map
 
-A getter function for checking the value of associated with a `minter` in the public `minters` mapping.
+A `map` is a state variable that "maps" a key to a value. It can be used with private or public storage variables.
 
-#include_code is_minter /noir-projects/noir-contracts/contracts/token_contract/src/main.nr rust
+:::info
+In Aztec.nr, keys are always `Field`s, or types that can be serialized as Fields, and values can be any type - even other maps. `Field`s are finite field elements, but you can think of them as integers.
+:::
 
-#### `total_supply`
+It includes a [`Context`](../../writing_contracts/functions/context.md) to specify the private or public domain, a `storage_slot` to specify where in storage the map is stored, and a `start_var_constructor` which tells the map how it should operate on the underlying type. This includes how to serialize and deserialize the type, as well as how commitments and nullifiers are computed for the type if it's private.
 
-A getter function for checking the token `total_supply`.
+You can view the implementation in the Aztec.nr library [here](https://github.com/AztecProtocol/aztec-packages/tree/master/noir-projects/aztec-nr).
 
-#include_code total_supply /noir-projects/noir-contracts/contracts/token_contract/src/main.nr rust
+A `map` can hold multiple different types of note types, due to note type IDs. These are identifiers for each note type that are unique within a contract.
 
-#### `balance_of_private`
+### `new`
 
-A getter function for checking the private balance of the provided Aztec account. Note that the [Private Execution Environment (PXE)](https://github.com/AztecProtocol/aztec-packages/tree/#include_aztec_version/yarn-project/pxe) must have access to the `owner`s decryption keys in order to decrypt their notes.
+When declaring the storage for a map, we use the `Map::new()` constructor. As seen below, this takes the `storage_slot` and the `start_var_constructor` along with the [`Context`](../../writing_contracts/functions/context.md).
 
-#include_code balance_of_private /noir-projects/noir-contracts/contracts/token_contract/src/main.nr rust
+We will see examples of map constructors for public and private variables in later sections.
 
-#### `balance_of_public`
+#### As private storage
 
-A getter function for checking the public balance of the provided Aztec account.
+When declaring a mapping in private storage, we have to specify which type of Note to use. In the example below, we are specifying that we want to use the `Singleton` note type.
 
-#include_code balance_of_public /noir-projects/noir-contracts/contracts/token_contract/src/main.nr rust
+In the Storage struct:
 
-#### `compute_note_hash_and_nullifier`
+#include_code storage-map-singleton-declaration /noir-projects/noir-contracts/contracts/docs_example_contract/src/main.nr rust
 
-A getter function to compute the note hash and nullifier for notes in the contract's storage.
+In the `Storage::init` function:
```

Comment:
Do we need this reference anymore, since the Storage::init has been abstracted away? I think we can remove it.